yt_dlp/extractor/nebula.py

```diff
@@ -179,60 +113,173 @@ def _fetch_zype_access_token(self, video_id):
+class NebulaCollectionIE(NebulaBaseIE):
+    IE_NAME = 'nebula:collection'
+    _VALID_URL = r'https?://(?:www\.)?(?:watchnebula\.com|nebula\.app)/(?!videos/)(?P<id>[-\w]+)'
+    _TESTS = [
+        {
+            'url': 'https://nebula.app/tom-scott-presents-money',
+            'info_dict': {
+                'id': 'tom-scott-presents-money',
+                'title': 'Tom Scott Presents: Money',
+                'description': 'Tom Scott hosts a series all about trust, negotiation and money.',
+            },
+            'playlist_count': 5,
+            'params': {
+                'usenetrc': True,
+            },
+        }, {
+            'url': 'https://nebula.app/lindsayellis',
+            'info_dict': {
+                'id': 'lindsayellis',
+                'title': 'Lindsay Ellis',
+                'description': 'Enjoy these hottest of takes on Disney, Transformers, and Musicals.',
+            },
+            'playlist_mincount': 100,
+            'params': {
+                'usenetrc': True,
+            },
+        },
+    ]
 
-        channel_title = self._extract_channel_title(video_meta)
+    def _fetch_collection(self, collection_id):
+        page_nr = 1
+        next_url = f'https://content.watchnebula.com/video/channels/{collection_id}/'
+        episodes = []
+        channel_details = None
+        while next_url:
+            channel = self._call_nebula_api(next_url, collection_id, auth_type='bearer',
+                                            note=f'Retrieving channel page {page_nr}')
+            if not channel_details:
+                channel_details = channel['details']
+            episodes.extend(channel['episodes']['results'])
+            next_url = channel['episodes']['next']
+            page_nr += 1
 
-        return {
-            'id': video_id,
-            'display_id': display_id,
-            '_type': 'url_transparent',
-            'ie_key': 'Zype',
-            'url': 'https://player.zype.com/embed/%s.html?access_token=%s' % (video_id, zype_access_token),
-            'title': video_meta.get('title'),
-            'description': video_meta.get('description'),
-            'timestamp': parse_iso8601(video_meta.get('published_at')),
-            'thumbnails': [{
-                'id': tn.get('name'),  # this appears to be null
-                'url': tn['url'],
-                'width': tn.get('width'),
-                'height': tn.get('height'),
-            } for tn in video_meta.get('thumbnails', [])],
-            'duration': video_meta.get('duration'),
-            'channel': channel_title,
-            'uploader': channel_title,  # we chose uploader = channel name
-            # TODO: uploader_url, channel_id, channel_url
-        }
+        return channel_details, episodes
+
+    def _real_extract(self, url):
+        collection_id = self._match_id(url)
+        channel_details, episodes = self._fetch_collection(collection_id)
```

Comment:
It is better if the episode list is a generator. It makes sure only the necessary API calls are made when used with `--playlist-end` and similiar. But in this case, having to get the `channel_details` makes it a bit tricky. It can be done with itertools/LazyList, but it's hard for me to explain how over a review. Can u make this a generator? If not, mind if I make the changes directly?

Reply:
[Here](https://github.com/yt-dlp/yt-dlp/pull/1694/commits/32e3d56af9a54bc1ead9d21fe639982dcbc47e1c)'s my suggestion for a generator implementation.

It is making one HTTP request upfront anyway, but I don't see a way around that assuming that `_real_extract` cannot delay returning the playlist metadata somehow.

The generator function isn't really pretty, but I couldn't figure out a way to unwind it so that there isn't either a break in the middle of the loop or some awkward code repetition — I went for the latter, which trades some conciseness for quicker understandability, in my opinion.

Reply:
If you don't like this implementation, yes, you may make the changes directly.